src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -2961,6 +2970,52 @@ NEVER_INLINE void InterpreterSlowPath::initializeGlobalVariable(ExecutionState&
     ASSERT_NOT_REACHED();
 }
 
+NEVER_INLINE void InterpreterSlowPath::createObjectOperation(ExecutionState& state, CreateObject* code, ByteCodeBlock* byteCodeBlock, Value* registerFile)
+{
+    if (code->m_dataRegisterIndex != REGISTER_LIMIT) {
+        CreateObjectPrepare::CreateObjectData* data = reinterpret_cast<CreateObjectPrepare::CreateObjectData*>(registerFile[code->m_dataRegisterIndex].payload());
+        ASSERT(data->m_fillIndex == data->m_properties.size());
+        Object* obj = registerFile[code->m_registerIndex].asObject();
+        obj->m_values = std::move(data->m_values);
+        obj->m_structure = ObjectStructure::create(state.context(), std::move(data->m_properties));
+    } else {
+        registerFile[code->m_registerIndex] = new Object(state);
+    }
+#if defined(ESCARGOT_SMALL_CONFIG)
+    registerFile[code->m_registerIndex].asObject()->markThisObjectDontNeedStructureTransitionTable();
+#endif
+}
+
+NEVER_INLINE void InterpreterSlowPath::createObjectPrepareOperation(ExecutionState& state, CreateObjectPrepare* code, ByteCodeBlock* byteCodeBlock, Value* registerFile)
+{
+    if (code->m_stage == CreateObjectPrepare::Init) {
+        CreateObjectPrepare::CreateObjectData* data = new CreateObjectPrepare::CreateObjectData(code->m_propertyCount, new Object(state));
+        registerFile[code->m_dataRegisterIndex] = Value(Value::FromPayload, reinterpret_cast<intptr_t>(data));
+        registerFile[code->m_objectIndex] = data->m_target;
+    } else {
+        ASSERT(code->m_stage == CreateObjectPrepare::FillKeyValue);
+        CreateObjectPrepare::CreateObjectData* data = reinterpret_cast<CreateObjectPrepare::CreateObjectData*>(registerFile[code->m_dataRegisterIndex].payload());
+
+        if (code->m_hasAtomicString) {
+            data->m_properties[data->m_fillIndex] = ObjectStructureItem(ObjectStructurePropertyName(AtomicString::fromPayload(registerFile[code->m_keyIndex].asString())),
+                                                                        ObjectStructurePropertyDescriptor::createDataDescriptor(ObjectStructurePropertyDescriptor::AllPresent));
+        } else {
+            data->m_properties[data->m_fillIndex] = ObjectStructureItem(ObjectStructurePropertyName(state,
+                                                                                                    registerFile[code->m_keyIndex]),
+                                                                        ObjectStructurePropertyDescriptor::createDataDescriptor(ObjectStructurePropertyDescriptor::AllPresent));
+            data->m_hasNonAtomicPropertyName = true;
+        }
+
+        data->m_values[data->m_fillIndex] = registerFile[code->m_valueIndex];
+        data->m_fillIndex++;
+    }
+}
+
+NEVER_INLINE void InterpreterSlowPath::createArrayOperation(ExecutionState& state, CreateArray* code, ByteCodeBlock* byteCodeBlock, Value* registerFile)
+{
+    registerFile[code->m_registerIndex] = new ArrayObject(state, (uint64_t)code->m_length);
+}
```

Comment:
Why did you add a function for creating `Array` object here?
This method seems quite simple, so it might be better to embed this function body into the interpreter

Reply:
In my opinion, calling GC_MALLOC is quite complicated, so inlining the new expression into the interpreter is not very efficient.
It's only a small effort to reduce the size of the interpreter function.

Reply:
Oh,, I see